### ISSUE TITLE
linter(no-loss-of-precision): False Positive 1000000000000000128

### ISSUE THREAD

**Issue by magic-akari** [oxc-project/oxc#11436]
OXC playground: https://playground.oxc.rs/#eNpVj9FqwzAMRX9l6KmFMJLCYOR55HVj+wLFUYrBsYIkdy0l/167IYH6xZKQzr33Dg5aODT162tOn8d3485faTjUxzeogKG9g6RYPr1Fwyu0JokqCD7aVqvjmfbmNvUcts4Eo44sE7QjBqWlghlFSQoRQ+D/X7Ik8TuZ+oG6FJ15znrr+SyUdy/0g0JR12lGFPEnIte7wso0lDNlZ0B6qpuPnMErBzQavsgFFCz4TNrcOB7oTM+AFLEP9MdJHE0472qTj370m57jaMKhy9bL0YWkZ83pV+CyPACwRHjd

eslint playground:
https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tbG9zcy1vZi1wcmVjaXNpb246IFwiZXJyb3JcIiovXG5cbmNvbnN0IGEgPSAxMjM0NVxuY29uc3QgYiA9IDEyMy40NTZcbmNvbnN0IGMgPSAxMjNlMzRcbmNvbnN0IGQgPSAxMjMwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMFxuY29uc3QgZSA9IDB4MUZGRkZGRkZGRkZGRkZcbmNvbnN0IGYgPSA5MDA3MTk5MjU0NzQwOTkxXG5jb25zdCBnID0gOTAwN18xOTkyNTQ3NDA5XzkxXG5jb25zdCB4ID0gKDEwMDAwMDAwMDAwMDAwMDAxMjgpLnRvRml4ZWQoMCkiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

**Comment by camc314** [comment]
I've filed an issue on the eslint side for this, as I believe oxlint is correctly reporting an error here.

<img width="1101" alt="Image" src="https://github.com/user-attachments/assets/1184b275-e628-4e07-9e08-65ef64ea936a" />


Furthermore I get a TS diagnostic when I put this number into my editor

<img width="1157" alt="Image" src="https://github.com/user-attachments/assets/c1b70447-d2e2-4481-bfbf-7bfcd259575b" />

It looks like it's something related to that specific number in eslint? a digit less/more and the diagnostic is reported

**Comment by camc314** [comment]
Also, how did you run into this case out of interest?

**Comment by magic-akari** [comment]
I think the implementation in ESLint is correct. My example code includes `toFixed`; you can run it to verify.
I discovered this in a note within the ECMAScript Spec:
https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-number.prototype.tofixed

**Comment by camc314** [comment]
this feels very silly but basically the number gets coerced to `1000000000000000128` when performint operations on it?

```
> 1000000000000000120 - 1000000000000000128
0
```
